docs/design/debug-api.md

```diff
@@ -0,0 +1,349 @@
+10. `depth` -	int -	Current call depth
+11. `refund`	- int -	Refund counter
+12. `error` -	string -	Error message if any
+
+#### Example Request opcodeLogger
+```JSON
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_traceTransaction",
+  "params": 
+    [
+      "0x8fc90a6c3ee3001cdcbbb685b4fbe67b1fa2bec575b15b0395fea5540d0901ae",
+      {
+        "tracer": "opcodeLogger"
+      }
+    ]
+}
+```
+#### Example Response
+```JSON
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    gas: 85301,
+    returnValue: "",
+    structLogs: [{
+        depth: 1,
+        error: "",
+        gas: 162106,
+        gasCost: 3,
+        memory: null,
+        op: "PUSH1",
+        pc: 0,
+        stack: [],
+        storage: {}
+    },
+      /* snip */
+    {
+        depth: 1,
+        error: "",
+        gas: 100000,
+        gasCost: 0,
+        memory: ["0000000000000000000000000000000000000000000000000000000000000006", "0000000000000000000000000000000000000000000000000000000000000000", "0000000000000000000000000000000000000000000000000000000000000060"],
+        op: "STOP",
+        pc: 120,
+        stack: ["00000000000000000000000000000000000000000000000000000000d67cbec9"],
+        storage: {
+          0000000000000000000000000000000000000000000000000000000000000004: "8241fa522772837f0d05511f20caa6da1d5a3209000000000000000400000001",
+          0000000000000000000000000000000000000000000000000000000000000006: "0000000000000000000000000000000000000000000000000000000000000001",
+          f652222313e28459528d920b65115c16c04f3efc82aaedc97be59f3f377c0d3f: "00000000000000000000000002e816afc1b5c0f39852131959d946eb3b07b5ad"
+        }
+    }]
+  }
+```
+
+## Error Codes
+
+| Error Codes |        Error message        |                              Solution                               |
+| :---------: | :-------------------------: | :-----------------------------------------------------------------: |
+|    32000    |      Transaction not found.      | Occurs when user is trying to trace non-existing transaction. |
+|    32001    | Block hash not found. |          Occurs when user is trying to get modified accounts for non-existing block hash.           |
+|    32002    | Block number not found. |          Occurs when user is trying to get modified accounts for non-existing block number.           |
+
+## Limits
+1. Trying to get modified accounts for more than ex. 100 blocks. Env. variable can be `DEBUG_MODIFIED_ACCOUNTS_LIMIT`
+
+
+## Metric Capturing
+
+Capture metrics for the following:
+
+1. Log every call to all filter API method, as a total amount.
+2. Log every success or fail for each new API method, as a total amount.
+
+## Tests
+The following test cases should be covered but additional tests would be welcome.
+
+1. Overall functionality of the methods.
+2. Test different scenarios with all possible parameter combinations.
```

Comment:
Please be more specific.
Also no need to cover UT's here. Just high level acceptance tests to cover the happy paths and failures based in interface values and larger edge cases is sufficient.